### PR TITLE
feat: add `role` data source

### PR DESCRIPTION
**Pull request by iverberk**

### CODE REVIEW

**Comment by iverberk** [comment]
@madpah any chance you could review and let me know if this is ok? I'm planning on sending more resources shortly.

**Comment by madpah** [comment]
@iverberk - can you please look at the test failures?

**Comment by iverberk** [comment]
Yep, I'll take a look 👍

**Comment by iverberk** [comment]
@madpah the tests seem to failing on two things:

1. Doc generation. I'll fix that.
2. Unavailable provider configuration environment variables. I assumed that those env variables would be set in the CI/CD configuration, otherwise I don't know what to point them to. Can you clarify which IQ server URL needs to be used and how it should be set? Locally I just use a docker container.

**Comment by madpah** [comment]
@iverberk - thanks - I'll look at 2.

**Review by madpah — APPROVED** [review]

**Comment by iverberk** [comment]
Ok, I think the docs are fixed.

**Review by madpah — APPROVED** [review]

File: internal/provider/role_data_source_test.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Read testing
+			{
+				Config: providerConfig + `data "sonatypeiq_role" "role_by_name" {
+					name = "Developer"
+				}`,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("data.sonatypeiq_role.role_by_name", "name", "Developer"),
+					resource.TestCheckResourceAttr("data.sonatypeiq_role.role_by_name", "id", "1da70fae1fd54d6cb7999871ebdb9a36"),
```
Comment (iverberk) [inline internal/provider/role_data_source_test.go:36]: I hope this is a stable ID. If not, we'll have to create a dedicated role for TF provider testing. I would have loved to also create a role resource but the API does not seem to support role creation, so it'll have to be manual.

Comment (madpah) [reply internal/provider/role_data_source_test.go:36]: Yeah - that won't be stable @iverberk - for now, best to remove line 36 from the check.

Comment (madpah) [reply internal/provider/role_data_source_test.go:36]: Well - I wouldn't expect it to be stable, but I do appear to have the same ID in my install?

Let's try it as is 👍

Comment (iverberk) [reply internal/provider/role_data_source_test.go:36]: You beat me to it. I actually pushed a commit to remove it, but after your merge :-)

**Pull request merged**